charts/k8ssandra/templates/crd/batch_job.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-crd-upgrader-job-k8ssandra
+  labels: {{ include "k8ssandra.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "10"
+spec:
+  backoffLimit: 3
+  template:
+    metadata:
+      labels: {{ include "k8ssandra.labels" . | indent 8 }}
+    spec:
+      restartPolicy: OnFailure
+      serviceAccountName: {{ .Release.Name }}-crd-upgrader-k8ssandra
+      containers:
+        - name: crd-upgrade-job-k8ssandra
+          image: {{ .Values.client.image }}
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          args:
+            - -upgradecrds
+            - --targetVersion
+            - {{ .Chart.Version }}
```

Comment:
We have a bit of a chicken and egg problem here. I checked out your branch to test and upgraded from my local chart on the file system. That chart version is not installed. The latest version in the Helm repo is `1.2.0-20210514022645-da7547a5`. I temporarily changed the version to that so that the CRDs would be downloaded.